### ISSUE TITLE
Possibility to ignore certain .gitrepo's

### ISSUE THREAD

**Issue by jojoxd** [ingydotnet/git-subrepo#364]
Hello,

I've got a project like follows:

```
bundles/
    subrepo/
        package.json
        .gitrepo
node_modules/ (.gitignored)
    @org/subrepo-package/ (copy of bundles/subrepo)
        .gitrepo
vendor/ (.gitignored)
    org/subrepo/ (symlink to bundles/subrepo)
        .gitrepo
```

now the problem I've found is that the .gitrepo inside the node_modules is actually seen as an actual gitrepo, so it tries to update it as well:

```
>>> git fetch --no-tags --quiet ssh://git@host/org/subrepo.git master
>>> git rev-parse FETCH_HEAD^0
>>> git update-ref refs/subrepo/node_modules/@org/subrepo/fetch 582bb30e0234ee5438efe4d66678c552d8aef0c8
git-subrepo: There are new changes upstream, you need to pull first.
```

this happens after the original bundle has been pushed like so:

```
git subrepo push --all
```

it seems like the symlink keeps git from finding the one in composer, as it won't try to use that .gitrepo.

---

So here's my question;

Is there a possibility we can get a .gitrepo-ignore (or similar) to get these edge-cases solved?

--Joey Vos

**Comment by JacopKane** [comment]
I'm having the same issue due to symlinked monorepo structure under node_modules folder, therefore all my symlinked subrepo modules are being tried to push again after their real directories are getting pushed successfully.

Probably not using `--all` and specifying each subrepo dir would solve this for now. 

Although when your subrepo count is increasing it's quite problematic (something like defining and maintaining every 44 subrepo directories for a generic push action).

If there's an directory lookup exclusion filter (or even better, if `g subrepo push --all` was honouring `.gitignore` exclusion, and not looking at these paths) would solve this issue.

**Comment by grimmySwe** [comment]
I tested out a solution for this on branch issue/364. Now we check that .gitrepo is part of current repo before using it. Please test and see if it solves your problem.

**Comment by JacopKane** [comment]
Indeed it worked @grimmySwe ! thanks a lot for the quick fix and update

**Comment by admorgan** [comment]
This is fixed by SHA d34e6eae631a377479683bbe6cd4f853dba15a97
